src/rtf/include/rtf/TestAssert.h

```diff
@@ -82,6 +106,29 @@
                                             RTF_SOURCEFILE(),\
                                             RTF_SOURCELINE()), dynamic_cast<RTF::TestCase*>(this)) )
 
+/** Conditional assertion with throwing failure exception.
+ * \ingroup Test Assertions
+ * \param message Message to be reported as the detail of TestMessage
+ * \param condition If this condition evaluates to \c false then the
+ *                  test failed.
+ */
+#define RTF_ASSERT_FAIL_IF(condition, message) \
+    RTF_TEST_REPORT("WARNING: RTF_ASSERT_FAIL_IF(condition, message) "\
```

Comment:
It would be ideal if we could turn this in a compilation warning rather than a runtime warning. Let me know if it is not possible, in that case I will approve the PR as it is. : )